apache/iotdb
# [IOTDB-1422] Support partial insert for new vector interfaces

- Resolve some bugs in PR #3300
- Support partial insert for new vector interfaces

test codes:
1. **create timeseries:**
```
private static void createAlignedTimeseries()
      throws StatementExecutionException, IoTDBConnectionException {
    List<String> measurements = new ArrayList<>();
    for (int i = 1; i <= 2; i++) {
      measurements.add("s" + i);
    }
    List<TSDataType> dataTypes = new ArrayList<>();
    dataTypes.add(TSDataType.INT64);
    dataTypes.add(TSDataType.INT32);
    List<TSEncoding> encodings = new ArrayList<>();
    for (int i = 1; i <= 2; i++) {
      encodings.add(TSEncoding.RLE);
    }
    session.createAlignedTimeseries(
        ROOT_SG1_D1_VECTOR1, measurements, dataTypes, encodings, CompressionType.SNAPPY, null);
  }
```

2. **insert records:**
```
private static void insertAlignedRecord()
      throws IoTDBConnectionException, StatementExecutionException {
    List<String> measurements = new ArrayList<>();
    List<TSDataType> types = new ArrayList<>();
    measurements.add("s1");
    measurements.add("s2");
    types.add(TSDataType.INT64);
    types.add(TSDataType.INT64); // s2 is mismatched

    for (long time = 0; time < 1; time++) {
      List<Object> values = new ArrayList<>();
      values.add(1L);
      values.add(2L);
      session.insertRecord(ROOT_SG1_D1_VECTOR1, time, measurements, types, values, true);
    }
  }
```
We should get error message:
```
Exception in thread "main" org.apache.iotdb.rpc.StatementExecutionException: 313: failed to insert measurements [s1, s2] caused by DataType mismatch, Insert measurement s2 type INT64, metadata tree type INT32
```
